sst/SimengMemInterface.cc

```diff
@@ -0,0 +1,279 @@
+#include <sst/core/sst_config.h>
+#include "SimengMemInterface.hh"
+
+#include <iostream>
+
+using namespace SST::SSTSimeng;
+
+SimengMemInterface::SimengMemInterface(StandardMem* mem, uint64_t cl, uint64_t max_addr, SST::Output* out): simeng::MemoryInterface() {
+    this->mem = mem;
+    this->clw= cl;
+    this->max_addr_memory = max_addr;
+    this->output = out;
+};
+
+void SimengMemInterface::sendProcessImageToSST(const span<char> image) {
```

Comment:
Yes, will be handing that in a separate PR.

Reply:
Just a quick note about this, even after the mem-optimisation fix. This needs to be done because the process image still needs to be constructed inside LinuxProcess. After the initial copy is completed I can then free the processImage reference inside SimEng. Unless you want something else?